pyaedt/filtersolutions_core/optimization_goals_table.py

```diff
@@ -0,0 +1,411 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import csv
+from ctypes import POINTER
+from ctypes import byref
+from ctypes import c_char_p
+from ctypes import c_int
+from ctypes import create_string_buffer
+from enum import Enum
+
+import pyaedt
+
+
+class OptimizationGoalParameter(Enum):
+    """Enumeration of optimization goals parameters table.
+
+    **Attributes:**
+
+    - RATIO: Represents transmission zeros ratio table.
+    - BANDWIDTH: Represents transmission zeros bandwidth table.
+    - LOWER_FREQUENCY = Represents lower frequency parameters.
+    - UPPER_FREQUENCY = Represents upper frequency parameters.
+    - GOAL_VALUE = Represents goal value parameters.
+    - CONDITION = Represents condition parameters.
+    - PARAMETER_NAME = Represents name of parameters.
+    - WEIGHT = Represents weight parameters.
+     -ENABLED = Represents status of using the goal parameters.
+    """
+
+    LOWER_FREQUENCY = 0
+    UPPER_FREQUENCY = 1
+    GOAL_VALUE = 2
+    CONDITION = 3
+    PARAMETER_NAME = 4
+    WEIGHT = 5
+    ENABLED = 6
+
+
+class OptimizationGoalsTable:
+    """Manipulates access to the entries of optimization goals table.
+
+    This class lets you to enter, edit or remove goal parameters in the optimization goals table.
+    The table includes the lower frequency, upper frequency, goal desired value, goal condition,
```

Comment:
Should we have any more details about these fields? We want the user to know what they are, what the valid values are, and how to use them.